includes/class-wc-payments-payment-method-messaging-element.php

```diff
@@ -98,11 +101,13 @@ public function init() {
 						'amount'   => WC_Payments_Utils::prepare_amount( $price, $currency_code ),
 						'currency' => $currency_code,
 					];
+
+					$product_price = $product_variations['base_product']['amount'];
 				}
 			}
 		}
 
-		$enabled_upe_payment_methods = $this->gateway->get_payment_method_ids_enabled_at_checkout();
+		$enabled_upe_payment_methods = $this->gateway->get_upe_enabled_payment_method_ids();
```

Comment:
This just ensures we're getting all of the enabled BNPL payment methods. `get_payment_method_ids_enabled_at_checkout()` is a filtered list for checkout. This means, if you visit a PDP while you have products in the cart, you might not see the correct BNPL methods in the PMME. For example, with a $5 product in the cart, a $60 PDP won't display Affirm ($50 minimum) because it's been filtered out based on cart totals. Demo [here](https://github.com/Automattic/woocommerce-payments/issues/9456#issuecomment-2438774856), second example from the bottom. This change solves that.